doc/smart-switch/PA-Validation/SmartSwitchPAValidationOffload.md

```diff
@@ -0,0 +1,172 @@
+- [2 Modules Design](#2-modules-design)
+  - [2.1 Dash Offload Manager](#21-dash-offload-manager)
+    - [2.1.1 SONiC GNMI server changes](#211-sonic-gnmi-server-changes)
+  - [2.2 PA Validation Offload](#22-pa-validation-offload)
+    - [2.2.1 PA Validation Offload ACL configuration](#221-pa-validation-offload-acl-configuration)
+  - [2.3 DPU Shut/Restart](#23-dpu-shutrestart)
+- [3. Test Plan](#3-test-plan)
+
+###### Revision
+
+| Rev |     Date    |       Author          | Change Description                |
+|:---:|:-----------:|:---------------------:|-----------------------------------|
+| 0.1 |  06/10/2024 |     Kumaresh Perumal  | Initial version                  |
+| 0.2 |  08/21/2024 |          Yakiv Huryk  | Added DashOffloadManager         |
+
+
+# About this Manual
+This document provides general information about offloading PA validation feature to NPU in Smartswitch.
+
+# Definitions/Abbrevations
+
+|                          |                                          |
+|--------------------------|------------------------------------------|
+| ACL                      | Access Control List                      |
+| NPU                      | Network Processing Unit                  |
+| DPU                      | Data Processing Unit                     |
+ 
+
+# 1. Requirements Overview
+
+## 1.1 Functional Requirements
+- Provide a generic infrastructure for offloading DASH functionality to the NPU (DashOffloadManager).
+- DashOffloadManager should listen to the DASH configuration and either intercept it for offloading, or forward it to the DPU.
+- Provide PA validation support in NPU when this feature is not supported in DPU pipeline model.
+- Use existing ACL tables and ACL rules design in NPU.
+
+## 1.2 Scale Requirements
+PA Validation configuration:
+* 4096 PA Validation entries
+
+Please refer to [SONiC-DASH HLD](https://github.com/sonic-net/SONiC/blob/master/doc/dash/dash-sonic-hld.md#3213-underlay-src-ip-pa-validation) for more details regarding the PA Validation configuration requirements
+
+ACL:
+- One ACL table per each DPU
+- 4160 ACL rules:
+  - 4096 ACL forward rules (one per each PA Validation address)
+  - 64 ACL drop rules (one per each unique VNI in the configuration)
+
+# 2 Modules Design
+
+## 2.1 Dash Offload Manager
+The new orchagent application DashOffloadManager will be responsible for DASH offloading logic. It will collect all the needed information for offloading and perform all the relevant configurations.
+To get the DASH configuration that should be offloaded, the DashOffloadManager will act as a transparent ZMQ proxy between the GNMI server and the DPU swss, forwarding all the configuration and intercepting the tables that should be offloaded.
+To simplify the management of the DPU offload and achieve optimal performance, every DPU is handled by a separate instance of a ZMQ Proxy (pair of ZMQ Server and Client)
+
+<img src="images/DashOffloadManager.svg">

```

Comment:
I assume this is after the gnmi splitter, will be better to make it more clear how it works with the independent dpu upgrade changes.